### ISSUE TITLE
Add an initial round of tests

### ISSUE THREAD

**Issue by alessandrofelder** [brainglobe/cellfinder-napari#31]
Currently we only have[ a placeholder test.](https://github.com/brainglobe/cellfinder-napari/blob/641f729a915cebe30a437f7a553e4eec3cb8d3b6/cellfinder_napari/_tests/test_cellfinder_napari.py)

Test for cellfinder-napari should focus on UI testing and workflow testing (actual computations done and tested in cellfinder-core)
Workshop of 19th January by napari devs should give us some additional guidance/tips to write tests.

Still need to decide on what exactly test contents will be.

**Comment by alessandrofelder** [comment]
# A possible strategy for initial round tests

Given that we should trust that `napari` widgets work as expected and `cellfinder-core` works as expected (and if not, we should raise an issue there!), the aspects of `cellfinder-napari` that need testing are (at least as an initial step):

* user inputs are validated correctly and expected error messages are thrown (#37 )
* progress bars show correctly and reaches end (e.g. #38 )
* maybe an integration test with some example data (~ check that hitting run on `cellfinder-napari` with the default parameters gives the same result as calling `cellfinder-core` itself with the same data and parameters)

@paddyroddy @dpshelio @adamltyson feedback on the above appreciated. What did I forget?

**Comment by paddyroddy** [comment]
Dare I say, some unit tests? I know it's less critical but probably worth the effort

**Comment by alessandrofelder** [comment]
Sure, what kind of things would said unit tests be testing? I guess some of the helper functions we already have (and we'll probably create more)?

**Comment by paddyroddy** [comment]
yeah just test each function is doing what we expect. Usual stuff 🤷

**Comment by dstansby** [comment]
https://napari.org/plugins/test_deploy.html is a useful resource for writing tests for plugins
